### PR TITLE
Fix clang warnings

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

There are six warnings reported by the CLANG builds which seems to be legit 
https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/slc7_amd64_gcc900/www/sun/12.0.CLANG-sun-23/CMSSW_12_0_CLANG_X_2021-05-30-2300
Not sure if the change for RecMuon->eta() was originaly meant to be over fabs(RecMuon->eta()) only 

#### PR validation:

Builds with the CLANG IB without the warnings

### CODE REVIEW

**Comment by mrodozov** [comment]
test parameters:
- release/arch = CMSSW_12_0_CLANG_X_2021-05-30-2300/slc7_amd64_gcc900

**Comment by mrodozov** [comment]
ping

**Comment by mrodozov** [comment]
please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaLowPUHBHEMuonFilter.cc [hunk 1]
```diff
@@ -179,7 +179,7 @@ bool AlCaLowPUHBHEMuonFilter::filter(edm::Event& iEvent, edm::EventSetup const&
               << RecMuon->innerTrack().isNonnull() << " outerTrack " << RecMuon->outerTrack().isNonnull()
               << " globalTrack " << RecMuon->globalTrack().isNonnull();
 #endif
-          if ((RecMuon->pt() < minimumMuonpT_) || fabs(RecMuon->eta() < minimumMuoneta_))
+          if ((RecMuon->pt() < minimumMuonpT_) || (RecMuon->eta() < minimumMuoneta_))
```
Comment (slava77) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaLowPUHBHEMuonFilter.cc:182]:
```suggestion
          if ((RecMuon->pt() < minimumMuonpT_) || std::abs(RecMuon->eta()) < minimumMuoneta_)
```
this seems more correct in this context

**Comment by mrodozov** [comment]
please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaLowPUHBHEMuonFilter.cc (hunk 1 above)
Comment (mrodozov) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaLowPUHBHEMuonFilter.cc]: as the comment says :) (but I wasn't sure)

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
hm that fails exactly in recotracker :/

**Comment by mrodozov** [comment]
please test
I think this will be consistent, lets see

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33912	cae8ad2
- changes in reco are technical (removing a fabs on uint values)

**Comment by yuanchao** [comment]
+1

* one of the check still in pending
* fix six warnings by CLANG builds (fabs -> std::abs, technical)

**Comment by mrodozov** [comment]
please test for CMSSW_12_0_CLANG_X/slc7_amd64_gcc900

**Comment by mrodozov** [comment]
lets get this comparison finished for CLANG

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**